inspectit-gepard-agent/src/main/java/rocks/inspectit/gepard/agent/internal/identity/IdentityManager.java

```diff
@@ -0,0 +1,59 @@
+/* (C) 2024 */
+package rocks.inspectit.gepard.agent.internal.identity;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Objects;
+import rocks.inspectit.gepard.agent.internal.identity.model.IdentityInfo;
+
+/** Responsible for generating the agentId. */
+public class IdentityManager {
+
+  private static IdentityManager instance;
+
+  private final IdentityInfo identityInfo;
+
+  private IdentityManager() {
+    RuntimeMXBean runtime = ManagementFactory.getRuntimeMXBean();
+    String pidAtHostname = runtime.getName();
+    this.identityInfo =
+        new IdentityInfo(runtime.getPid(), pidAtHostname.split("@")[1], hash(pidAtHostname));
```

Comment:
Couldn't we just use one property in `IdentityInfo` for the JVM name instead of two for pid and hostname?
```
String jvmName = runtime.getName();
this.identityInfo = new IdentityInfo(jvmName, hash(jvmName));
```


I'm a bit worried that splitting at "@" might not always work, because the output of `getName()` depends on the JVM implementation. It might differ in other implementations...

Reply:
👍🏻